pkg/mobile/bindableServices.go

```diff
@@ -48,15 +89,32 @@ func (lister *BindableMobileServiceCRUDLImpl) Delete(namespace string, bindingNa
 	return lister.scClient.ServiceBindings(namespace).Delete(bindingName, &v1.DeleteOptions{})
 }
 
-func (lister *BindableMobileServiceCRUDLImpl) Watch(namespace string, mobileClientName string) func() (watch.Interface, error) {
+func (lister *BindableMobileServiceCRUDLImpl) Watch(mobileClient *v1alpha1.MobileClient) func() (watch.Interface, error) {
 	return func() (watch.Interface, error) {
 		watchOpts := metav1.ListOptions{}
-		return lister.scClient.ServiceBindings(namespace).Watch(watchOpts)
+		return lister.scClient.ServiceBindings(mobileClient.GetNamespace()).Watch(watchOpts)
 	}
 }
 
-func (lister *BindableMobileServiceCRUDLImpl) List(namespace string, mobileClientName string) (*BindableMobileServiceList, error) {
+func (bms *BindableMobileServiceCRUDLImpl) DeleteAppData(mobileClient *v1alpha1.MobileClient) error {
+	bindListOpts := v1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", MOBILE_CLIENT_ID_LABEL_NAME, mobileClient.GetName()),
+	}
+
+	bindings, err := bms.scClient.ServiceBindings(mobileClient.GetNamespace()).List(bindListOpts)
+
+	if err != nil {
+		return err
+	}
+
+	for _, binding := range bindings.Items {
+		bms.Delete(mobileClient.GetNamespace(), binding.GetName())
```

Comment:
Not sure about this, but should we check for errors here? In this code [1] we return if there is an error - I thought that that is so that we won't delete an app if some of it's resources can not be deleted.

[1] https://github.com/aerogear/mobile-developer-console/pull/142/files#diff-891bcbc429e4f6564bef6d4f9d6abea4R21

Reply:
ok, I will fix this.